axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorderline/SaleOrderLineCreateTaxLineServiceImpl.java

```diff
@@ -136,15 +144,61 @@ protected SaleOrderLineTax createSaleOrderLineTax(
   protected void computeAndAddTaxToList(
       Map<TaxLine, SaleOrderLineTax> map,
       List<SaleOrderLineTax> saleOrderLineTaxList,
-      Currency currency) {
+      Currency currency,
+      List<SaleOrderLineTax> currentSaleOrderLineTaxList) {
     for (SaleOrderLineTax saleOrderLineTax : map.values()) {
       // Dans la devise de la facture
       orderLineTaxService.computeTax(saleOrderLineTax, currency);
-      saleOrderLineTaxList.add(saleOrderLineTax);
-      LOG.debug(
-          "VAT line : VAT total => {}, W.T. total => {}",
-          saleOrderLineTax.getTaxTotal(),
-          saleOrderLineTax.getInTaxTotal());
+      SaleOrderLineTax oldSaleOrderLineTax =
+          isMapValueAlreadyInSaleOrderLineTaxList(saleOrderLineTax, currentSaleOrderLineTaxList);
+      if (oldSaleOrderLineTax == null) {
+        saleOrderLineTaxList.add(saleOrderLineTax);
+        LOG.debug(
+            "VAT line : VAT total => {}, W.T. total => {}",
+            saleOrderLineTax.getTaxTotal(),
+            saleOrderLineTax.getInTaxTotal());
+      } else {
+        saleOrderLineTaxList.add(oldSaleOrderLineTax);
+      }
+    }
+  }
+
+  @Override
+  public List<SaleOrderLineTax> getUpdatedSaleOrderLineTax(SaleOrder saleOrder) {
+    List<SaleOrderLineTax> saleOrderLineTaxList = new ArrayList<>();
+
+    if (ObjectUtils.isEmpty(saleOrder.getSaleOrderLineTaxList())) {
+      return saleOrderLineTaxList;
     }
+
+    saleOrderLineTaxList.addAll(
+        saleOrder.getSaleOrderLineTaxList().stream()
+            .filter(
+                saleOrderLineTax ->
+                    orderLineTaxService.isManageByAmount(saleOrderLineTax)
+                        && saleOrderLineTax
+                                .getTaxTotal()
+                                .compareTo(saleOrderLineTax.getPercentageTaxTotal())
+                            != 0)
+            .collect(Collectors.toList()));
+    return saleOrderLineTaxList;
+  }
+
+  protected SaleOrderLineTax isMapValueAlreadyInSaleOrderLineTaxList(
```

Comment:
Please rename this method, `is...()` or `has...()` method names suggest that it would return a `boolean` by convention (which is not the case here). In this case it could be `getSaleOrderLIneTaxFromList`